synapse/tests/test_lib_stormtypes.py

```diff
@@ -5186,12 +5201,23 @@ async def test_stormtypes_tofoo(self):
         self.true(await s_stormtypes.tobool(boolprim))
         self.true(await s_stormtypes.tobool(1))
         self.false(await s_stormtypes.tobool(0))
+        self.true(await s_stormtypes.tobool(s_stormtypes.Number('1')))
+        self.false(await s_stormtypes.tobool(s_stormtypes.Number('0')))
         # no bool <- int <- str
         self.true(await s_stormtypes.tobool('1'))
         self.true(await s_stormtypes.tobool(s_stormtypes.Str('0')))
         self.true(await s_stormtypes.tobool(s_stormtypes.Str('asdf')))
         self.false(await s_stormtypes.tobool(s_stormtypes.Str('')))
 
+        self.eq(20, await s_stormtypes.tonumber('20'))
+        self.eq(20.1, await s_stormtypes.tonumber(20.1))
+        self.eq(20.1, await s_stormtypes.tonumber('20.1'))
+        self.eq(20.1, await s_stormtypes.tonumber(s_stormtypes.Number('20.1')))
+
+        self.eq(True, await s_stormtypes.tocmprvalu(boolprim))
+        self.eq((1, 3), await s_stormtypes.tocmprvalu([1, s_exc.SynErr, 3]))
```

Comment:
These test cases give me pause. Lets discuss :)

Reply:
updated to allow these through